substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3029,6 +3095,58 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	fn do_adjust_ed_deposit(who: T::AccountId, pool: PoolId) -> DispatchResult {
+		let bonded_pool = BondedPool::<T>::get(pool).ok_or(Error::<T>::PoolNotFound)?;
+		// only depositor can adjust ED deposit.
```

Comment:
we can change this in the future tbh, why not let anyone do it if they want to?

Reply:
Do you say this so the existing deficit is fixed? Because that will be fixed by the initial migration.

I am leaning towards it being permissioned since
- It also allows claiming back excess ED. That part cannot be permissionless.
- If someone else tops up the deficit, eventually it will be depositor who will get that deposit back.

Reply:
> It also allows claiming back excess ED. That part cannot be permissionless

Why not? This is actually a great incentive to make sure if there is excess, someone will claim it as soon as possible. I see your concern as well, but I don't think it is an issue. If anything, I think it is an advantage 😅